### ISSUE TITLE
XaAES xa_boot.log

### ISSUE THREAD

**Issue by latzmaster** [freemint/freemint#287]
File xa_boot.log is always written, setting option <logfile=""> in xaaes.cnf don´t work.

**Comment by xdelatour** [comment]
It is because XaAES starts writing to xa_boot.log before reading xaaes.cnf.

**Comment by mikrosk** [comment]
Helmut was fighting this quite fiercely, at first he wouldn't even let the user disable the logging _at all_. :) But I agree it should read the variable first and write later. Shouldn't be too hard to fix.

**Comment by th-otto** [comment]
Handling en empty logfile name should be too hard, yes. But identifying all log calls that are possibly made before xaaes.cnf is read, is not so easy i think.

**Comment by latzmaster** [comment]
Just an idea - what about (simply) adding a parameter to xaloader and/or xaaes.km?
Settable in mint.cnf then: INIT=.../xaloader.prg -v

**Comment by mikrosk** [comment]
@latzmaster yes, that had been considered back in 2013 as well. I'm not against it, it could work in a similar fashion as it does with freemint -- one could specify the output device (console, serial port, natfeats output etc) directly in that command line parameter and we could get rid of that log levels (what a misleading name...) and log files entirely.

**Comment by skaftetryne** [comment]
You can also remove this setting from xaaes.cnf completely, and move it to an environment variable instead. There are two advantages with this:

1. You can enable logging from mint.cnf, and XaAES will be able to log from the start.
2. External tools (like a log viewer) will know where to find the log file.

It is also very easy to implement.

**Comment by mikrosk** [comment]
Btw @th-otto, with #32 being done, the @skaftetryne's suggestion literally _begs_ to implemented as well. ;)

(i.e. an environment variable in xaaes.cnf which would use $SYSDIR, e.g. `setenv XAAES_LOGFILE ${SYSDIR}/xaaes/xaaes.log`

**Comment by mikrosk** [comment]
Implemented exactly as @skaftetryne suggested.
